src/mlpack/core/data/load_impl.hpp

```diff
@@ -20,184 +21,233 @@
 #include <exception>
 
 #include "extension.hpp"
-#include "detect_file_type.hpp"
+#include "load_utilities.hpp"
 #include "string_algorithms.hpp"
 
+
 namespace mlpack {
 namespace data {
 
-namespace details{
-
-template<typename Tokenizer>
-std::vector<std::string> ToTokens(Tokenizer& lineTok)
+template<typename MatType>
+bool LoadCSVASCII(const std::string& filename,
+                  MatType& matrix,
+                  const DataOptions& opts)
```

Comment:
Why move this function into this file?  It feels like it would be better to have left it in `load_numeric_csv.hpp`, or at least some separate file.  These load functions are complicated so it probably makes sense to keep them separate.